### PR TITLE
Fixed MultiAtlas load with filter, and added tests

### PR DESCRIPTION
**Pull request by adahn6**
### Description:

This PR includes a fix for MultiAtlas.loadFromPackedAtlas() for loading over an Iterable of Resources with a Predicate filter. In cases in which the Predicate filters out all entities from a valid Atlas Resource in the Iterable, the result is an empty Optional; the previous code simply invokes .get() without checking, causing an error. This fix filters out all empty Optionals from the filtered Atlas Resources before loading, removing this error. 

Additionally, a few new tests have been added around this behavior and other [recent MultiAtlas behavior changes](https://github.com/osmlab/atlas/pull/408). As part of that change, lots of tests that were depending on PackedAtlas.getAtlas() and other Atlases built directly through PackedAtlasBuilder in test methods have been updated to have their own Rules with TestAtlases defined. 

Lastly, TestAtlas.Relation was augmented to allow for an OSM identifier to be specified-- this allows a synthetically sliced relation to be easily defined in TestAtlas annotations. Without being specified, it will default to the same identifier as the Relation.

### Potential Impact:
None, as this is just one bug fix and some tests were changed.

### Unit Test Approach:
Two unit tests capturing behavior previously not tested for, additionally reworked and streamlined test definitions.

### Test Results:
Describe other (non-unit) test results here.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/test/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlasTestRule.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Loc;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Node;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Point;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Relation;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Relation.Member;
+
+/**
+ * @author samg
+ */
+public class PackedAtlasTestRule extends CoreTestRule
```
Comment (lucaspcram) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlasTestRule.java]: Much appreciated updates. This makes the tests easier to understand.

File: src/main/java/org/openstreetmap/atlas/geography/atlas/multi/MultiAtlas.java [hunk 2]
```diff
@@ -192,12 +192,13 @@ public static MultiAtlas loadFromPackedAtlas(final Iterable<? extends Resource>
...
-                        .translate(resources,
-                                resource -> PackedAtlas.load(resource)
-                                        .subAtlas(filter, AtlasCutType.SOFT_CUT).get()),
-                lotsOfOverlap);
+        final Iterable<Atlas> validAtlases = Iterables
+                .translate(Iterables
+                        .stream(Iterables.translate(resources,
+                                resource -> PackedAtlas.load(resource).subAtlas(filter,
+                                        AtlasCutType.SOFT_CUT)))
+                        .filter(optional -> optional.isPresent()), optional -> optional.get());
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/multi/MultiAtlas.java]: Small nit:
`Optional::isPresent` and `Optional::get`

Comment (adahn6) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/multi/MultiAtlas.java]: good call!

File: src/test/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlasTest.java [hunk 3]
```diff
@@ -56,132 +51,13 @@
...
-        builder.addPoint(6, Location.TEST_1, pointTags);
-        builder.addPoint(7, Location.TEST_5, pointTags);
-
-        // Add relations
-        builder.addRelation(1, 1, structure1, relationTags);
-        builder.addRelation(2, 2, structure2, relationTags);
-
-        // Final call
-        this.atlas = (PackedAtlas) builder.get();
+        this.atlas = this.setup.getAtlas().cloneToPackedAtlas();
```
Comment (matthieun) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlasTest.java]: Nice!

**Pull request merged**